source/declarations.tex

```diff
@@ -6091,6 +6098,223 @@
 \end{example}%
 \indextext{definition!function|)}
 
+\rSec2[dcl.fct.def.coroutine]{Coroutine definitions}%
+\indextext{definition!coroutine}%
+
+\pnum
+A function is a \defn{coroutine} if it contains a
+\grammarterm{coroutine-return-statement}\iref{stmt.return.coroutine},
+an \grammarterm{await-expression}\iref{expr.await},
+or a \grammarterm{yield-expression}\iref{expr.yield}.
+The \grammarterm{parameter-declaration-clause} of the coroutine shall not
+terminate with an ellipsis that is not part of
+a \grammarterm{parameter-declaration}.
+
+\pnum
+\begin{example}
+\begin{codeblock}
+task<int> f();
+
+task<void> g1() {
+  int i = co_await f();
+  std::cout << "f() => " << i << std::endl;
+}
+
+template <typename... Args>
+task<void> g2(Args&&...) { // OK: ellipsis is a pack expansion
+  int i = co_await f();
+  std::cout << "f() => " << i << std::endl;
+}
+
+task<void> g3(int a, ...) { // error: variable parameter list not allowed
+  int i = co_await f();
+  std::cout << "f() => " << i << std::endl;
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+For a coroutine \textit{f} that is a non-static member function, let
+$\placeholder{P}_1$ denote the type of the implicit object
+parameter\iref{over.match.funcs} and
+$\placeholder{P}_2 \dotsc \placeholder{P}_n$ be the types of the function
+parameters;
+otherwise let $\placeholder{P}_1 \dotsc \placeholder{P}_n$ be the types of
+the function parameters.
+Let $\placeholder{p}_1 \dotsc \placeholder{p}_n$ be lvalues denoting those
+objects.
+Let \placeholder{R} be the return type and \placeholder{F} be the
+\grammarterm{function-body} of \placeholder{f}, \placeholder{T} be the type
+\tcode{std::coroutine_traits<$\placeholder{R}$, $\placeholder{P}_1$, $\dotsc$, $\placeholder{P}_n$>},
+and \placeholder{P} be the class type denoted by
+\placeholder{T}\tcode{::promise_type}.
```

Comment:
Hmm, this uses P-with-subscript for Parameter and P-without-subscript for Promise. Seems a bit unnecessary to use the letter P for these different things, might be clearer to use different letters. :)

Reply:
Great point! renamed to A_i and a_i by commit b0460b7 below

Reply:
I find this particularly counter-intuitive. There is a material difference between parameters and arguments, and we use P/A all over the place in overload resolution, so this should really stay "Pi" (because we're talking about parameter types here). In contrast, "promise type" is not yet used anywhere, so we can be a bit more creative with that. Maybe just "C"? Or "S"?

Reply:
Okay. Let me play with that. Do placeholder have to be single letter? 
What about _Promise_ and _promise_?

Reply:
I changed it to S and s, keep parameters to be P_i and p_i as in the TS